pkg/controller/template/template.go

```diff
@@ -0,0 +1,81 @@
+package template
+
+import (
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/goharbor/harbor-operator/pkg/config"
+	"github.com/ovh/configstore"
+	"github.com/pkg/errors"
+)
+
+const (
+	ConfigTemplateKey     = "template-content"
+	ConfigTemplatePathKey = "template-path"
+)
+
+type NotYetRefreshed struct {
+	error
+}
+
+type Template struct {
+	Path      string
+	Key       string
+	Priority  int64
+	refreshed bool
+}
+
+func New(path string) *Template {
+	return &Template{
+		Path:      path,
+		Key:       ConfigTemplateKey,
+		Priority:  config.DefaultPriority,
+		refreshed: false,
+	}
+}
+
+func FromConfigStore(store *configstore.Store, defaultFileName string) (*Template, error) {
+	templateDir, err := config.GetString(configstore.DefaultStore, config.TemplateDirectoryKey, config.DefaultTemplateDirectory)
+	if err != nil {
+		return nil, errors.Wrap(err, "directory")
+	}
+
+	defaultTemplatePath := path.Join(templateDir, defaultFileName)
+
+	configTemplatePath, err := config.GetString(store, ConfigTemplatePathKey, defaultTemplatePath)
+	if err != nil {
+		return nil, errors.Wrap(err, "path")
+	}
+
+	fileStat, err := os.Stat(configTemplatePath)
+	if err != nil {
+		return nil, errors.Wrap(err, "stat")
+	}
+
+	if fileStat.IsDir() {
+		return nil, errors.Wrap(err, "not a file")
+	}
+
+	return New(configTemplatePath), nil
+}
+
+func (t *Template) RegisterProvider(store *configstore.Store, name string) {
+	store.FileCustomRefresh(t.Path, func(data []byte) ([]configstore.Item, error) {
+		t.refreshed = true
+
+		return []configstore.Item{configstore.NewItem(t.Key, string(data), t.Priority)}, nil
+	})
+}
+
+func (t *Template) ReadyzCheck(req *http.Request) error {
```

Comment:
Could be renamed to `Ready` and simply return a boolean

Reply:
This should match the [`healthz.Checker` signature](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.6.2/pkg/healthz?utm_source=gopls#Checker)

Reply:
Okay we should probably specify it in a comment then, as there is no other occurrence in the project it's not trivial to figure out.
```suggestion
// Should respect the healthz.Checker signature (from sigs.k8s.io/controller-runtime/pkg/healthz)
func (t *Template) ReadyzCheck(req *http.Request) error {
```

Reply:
I added the check at compilation using:

```golang
var _ healthz.Checker = (&Template{}).HealthzCheck
```